packages/dd-trace/src/instrumenter.js

```diff
@@ -207,6 +229,36 @@ class Instrumenter {
     }
   }
 
+  preload () {
+    if (!this._loader.preload) {
+      return
+    }
+    const pluginsMap = new Map()
+    Object.keys(plugins)
+      .filter(name => !this._plugins.has(plugins[name]))
+      .forEach(name => {
+        pluginsMap.set(plugins[name], { name, config: getConfig(name) })
+      })
+    this._loader.preload(pluginsMap)
+  }
+
+  prepatch (instrumentation, moduleExports) {
+    if (!instrumentation.prepatch) {
+      return
+    }
+    const patches = [].concat(instrumentation.prepatch(moduleExports))
+    for (const { object, methods } of patches) {
+      for (const name of methods) {
+        this.wrap(object, name, getPrepatchWrappedFunction)
+        // While we're re-using wrap here for simplicity, we don't want to have
+        // the other code here assume that it has been pached in the sense that
+        // the _plugin's_ `patch` function has been called. We'ere adding a
+        // `_datadog_wrapped` property instead.
+        delete object[name]._datadog_patched
```

Comment:
Why would this be there at that point?

Reply:
`this.wrap` always adds this property, as explained in the comment.